src/patch-place-break/patch-place-break-core/src/test/java/fr/djaytan/minecraft/jobsreborn/patchplacebreak/core/PatchPlaceBreakImplTest.java

```diff
@@ -0,0 +1,329 @@
+  @Captor private ArgumentCaptor<BlockLocation> blockLocationCaptor;
+  private final RestrictedBlocksProperties restrictedBlocksProperties =
+      RestrictedBlocksProperties.of(
+          new HashSet<>(Arrays.asList("STONE", "DIRT")), RestrictionMode.BLACKLIST);
+  private PatchPlaceBreakImpl patchPlaceBreakImpl;
+
+  @BeforeEach
+  void beforeEach() {
+    BlockSetFilter blockSetFilter = new BlockSetFilter(restrictedBlocksProperties);
+
+    this.patchPlaceBreakImpl =
+        new PatchPlaceBreakImpl(tagRepository, restrictedBlocksProperties, blockSetFilter);
+  }
+
+  @Nested
+  @DisplayName("When put tag")
+  class WhenPutTag {
+
+    @Test
+    @DisplayName("With not restricted block")
+    void withNotRestrictedBlock() {
+      // Given
+      BlockLocation blockLocation = BlockLocation.of("world", 0, 0, 0);
+      Block blockToTag = Block.of(blockLocation, "BEACON");
+
+      // When
+      patchPlaceBreakImpl.putTag(blockToTag, false);
+      verify(tagRepository, timeout(1000)).put(tagCaptor.capture());
+
+      // Then
+      Tag tag = tagCaptor.getValue(); // Récupère la valeur Tag donné au TagRepository
+
+      assertAll(
+          () -> assertThat(tag.getBlockLocation()).isEqualTo(blockLocation),
+          () -> assertThat(tag.isEphemeral()).isFalse()); // TODO : Faire ça dans les autres test
+    }
+
+    @Test
+    @DisplayName("With restricted block")
+    void withRestrictedBlock() {
+      // Given
+      BlockLocation blockLocation = BlockLocation.of("world", 0, 0, 0);
+      Block blockToTag = Block.of(blockLocation, "STONE");
+
+      // When
+      patchPlaceBreakImpl.putTag(blockToTag, false);
+
+      // Then
+      verifyNoInteractions(tagRepository);
+    }
+
+    @Test
+    @DisplayName("With ephemeral tag")
+    void withEphemeralTag() {}
+  }
+
+  @Nested
+  @DisplayName("When remove tag")
+  class WhenRemoveTag {
+
+    @Test
+    @DisplayName("With not restricted block")
+    void withNotRestrictedBlock() {
+      // Given
+      BlockLocation newBlockLocation = BlockLocation.of("world", 0, 0, 0);
+      Block blockToRemoveTag = Block.of(newBlockLocation, "BEACON");
+
+      // When
+      patchPlaceBreakImpl.removeTag(blockToRemoveTag);
+      verify(tagRepository, timeout(1000)).delete(blockLocationCaptor.capture());
+
+      // Then
+      BlockLocation capturedBlockLocation = blockLocationCaptor.getValue();
+      assertThat(capturedBlockLocation).isEqualTo(newBlockLocation);
+    }
+
+    @Test
+    @DisplayName("With restricted block")
+    void withRestrictedBlock() {
+      // Given
+      BlockLocation blockLocation = BlockLocation.of("world", 0, 0, 0);
+      Block blockToTag = Block.of(blockLocation, "STONE");
+
+      // When
+      patchPlaceBreakImpl.removeTag(blockToTag);
+
+      // Then
+      verifyNoInteractions(tagRepository);
+    }
+  }
+
+  @Nested
+  @DisplayName("When move tags")
+  class WhenMoveTags {
+
+    @Test
+    @DisplayName("With not restricted blocks")
```

Comment:
"With **only unrestricted** blocks"
D'une manière général il est plus élégant de remplacer "not restricted" par "unrestricted"

Reply:
Compris, merci

Done